backend/app.py

```diff
@@ -216,39 +233,63 @@ def get_cset_members_items(codeset_ids: List[int] = None, con=CON) -> List[Legac
         {'codeset_ids': codeset_ids})
 
 
-def hierarchy(codeset_ids: List[int] = None, selected_concept_ids: List[int] = None) -> Dict:
-    """Get hierarchy of concepts in selected concept sets"""
-    if not selected_concept_ids:
-        selected_concept_ids = get_concept_set_member_ids(codeset_ids, column='concept_id')
-
-    # selected_roots: List[int] = top_level_cids(selected_concept_ids)
+def get_parent_children_map(root_cids: List[int], cids: List[int], con=CON) -> Dict[int, List[int]]:
```

Comment:
@Sigfried `new_hierarchy()` -> `hierarchy_helper()` -> now called `get_parent_children_map()`

To solve our problem with `IN (ARRAY[...])`, I tried some other things other than string interpolation:

```python
params = {'root_cids': root_cids, 'cids': cids}
query = text(query_str)
query = query.bindparams(bindparam('root_cids', expanding=True))
query = query.bindparams(bindparam('cids', expanding=False))
query = query.bindparams(root_cids=tuple(root_cids), cids=tuple(cids))
```

However I got the same problem. I ended up sticking with plain string interpolation.